test/solidity/Facets/ChainflipFacet.t.sol

```diff
@@ -0,0 +1,472 @@
+        emit LiFiTransferStarted(bridgeData);
+
+        initiateBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function test_CanBridgeTokensToEthereum()
+        public
+        assertBalanceChange(
+            ADDRESS_USDC,
+            USER_SENDER,
+            -int256(defaultUSDCAmount)
+        )
+        assertBalanceChange(ADDRESS_USDC, USER_RECEIVER, 0)
+        assertBalanceChange(ADDRESS_DAI, USER_SENDER, 0)
+        assertBalanceChange(ADDRESS_DAI, USER_RECEIVER, 0)
+    {
+        // Set source chain to Arbitrum for this test
+        vm.chainId(CHAIN_ID_ARBITRUM);
+        vm.roll(208460950); // Set specific block number for Arbitrum chain
+
+        // Set destination to Ethereum
+        bridgeData.destinationChainId = CHAIN_ID_ETHEREUM;
+        validChainflipData = ChainflipFacet.ChainflipData({
+            dstToken: 3, // USDC on Ethereum
+            nonEVMReceiver: bytes32(0), // Not needed for EVM chains
+            message: "",
+            gasAmount: 0,
+            cfParameters: ""
+        });
+
+        vm.startPrank(USER_SENDER);
+
+        // approval
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+
+        //prepare check for events
+        vm.expectEmit(true, true, true, true, _facetTestContractAddress);
+        emit LiFiTransferStarted(bridgeData);
+
+        initiateBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function testRevert_WhenUsingUnsupportedDestinationChain() public {
+        // Set destination chain to Polygon (unsupported)
+        bridgeData.destinationChainId = 137;
+
+        vm.startPrank(USER_SENDER);
+
+        // approval
+        usdc.approve(_facetTestContractAddress, bridgeData.minAmount);
+
+        vm.expectRevert(ChainflipFacet.UnsupportedChainflipChainId.selector);
+
+        initiateBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function test_CanBridgeNativeTokensWithDestinationCall()
+        public
+        assertBalanceChange(
+            address(0),
+            USER_SENDER,
+            -int256(defaultNativeAmount)
+        )
+    {
+        // Set destination to Arbitrum where our receiver contract is
+        bridgeData.destinationChainId = CHAIN_ID_ARBITRUM;
+        bridgeData.hasDestinationCall = true;
+        bridgeData.sendingAssetId = address(0);
+        bridgeData.minAmount = defaultNativeAmount;
+
+        // Create swap data for the destination chain
+        LibSwap.SwapData[] memory destSwapData = new LibSwap.SwapData[](0);
+
+        // Encode the message for the receiver contract
+        bytes memory message = abi.encode(
+            bridgeData.transactionId,
+            destSwapData,
+            USER_RECEIVER
+        );
+
+        validChainflipData = ChainflipFacet.ChainflipData({
+            dstToken: 7,
+            nonEVMReceiver: bytes32(0),
+            message: message,
+            gasAmount: 100000,
```

Comment:
(how) is this amount being passed to the facet/diamond?
I think you would need to use "addToMessageValue" here, no?

Reply:
Gas amount needs to be calculated as a fee taken from the bridge amount. It isn't calculated as extra like in other bridges.